### ISSUE TITLE
Problem with undelegated test

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-gui#38]
I selected "undelegated test" and ran some tests:

1. A domain name and its NS names without IP addresses and clicked on "check". I could see in the data that is sent (JSON) that the "nameserver" array was empty, which is incorrect.

2. A domain name and its NS names without IP addresses and clicked "fetch data from parent" (which added the IP addresses of NS) and then "check". Either:

   1. Nothing happened.

   2. The test was executed, but the "nameserver" array was empty.

3. A domain name and its NS names **with** IP addresses and clicked on "check". Now it sent the "nameserver" array with data.

This is related to zonemaster/zonemaster-engine#530

**Comment by sandoche2k** [comment]
@mtoma @matsduf @pamasse @vlevigneron is this issue at GUI or backend or Engine level?

**Comment by pamasse** [comment]
@sandoche2k The issue is not in the GUI. The request send during this kind of test contain the NS information that must be reuse by the backend.
For afnic.fr: 
{"jsonrpc":"2.0","id":1555569583349,"result":{"ds_list":[{"algorithm":8,"digtype":2,"digest":"d2681e301f632bd76544e6d5b6631a12d97b5479ff07cd24efecd19203c77db3","keytag":37610}],"ns_list":[{"ns":"ns1.nic.fr","ip":"192.134.4.1"},{"ns":"ns1.nic.fr","ip":"2001:67c:2218:2::4:1"},{"ns":"ns2.nic.fr","ip":"192.93.0.4"},{"ns":"ns2.nic.fr","ip":"2001:660:3005:1::1:2"},{"ns":"ns3.nic.fr","ip":"192.134.0.49"},{"ns":"ns3.nic.fr","ip":"2001:660:3006:1::1:1"}]}}

**Comment by matsduf** [comment]
@pamasse  and @sandoche2k!

I have tested this on firefox and chrome on MacOS with the same result. The pictures are from firefox.

In the first test I tested domain "fi" and gave  bill.ip.se and bull.ip.se as NS with no IP addresses, which should be fine since they are not in-bailiwick. In [fi-with-ns-with-no-ip] you can see that the 
"nameservers" array is empty, i.e. the configuration is not sent by GUI to the backend.

In the second test I tested the same "fi" but this time I give the IP addresses for bill.ip.se and bull.ip.se. In [fi-with-ns-with-ip-addr] you can see that the array is populated this time.

In a third test I tested domain "fi" and gave bill and bull without IP again, but I pressed "fetch data from parent zone" which gave all the real NS and IP for "fi" (bill and bull are just fake), but the first line was left with bill with no IP, and then no array was sent. If I deleted the NS with no IP the GUI sent the array to the backend.

```
bill.ip.se.		A	46.21.96.58
bill.ip.se.		AAAA	2a02:750:12::53
bull.ip.se.		A	212.85.74.18
bull.ip.se.		AAAA	2001:470:28:5a0::53
```

[fi-with-ns-with-no-ip]: https://user-images.githubusercontent.com/5430733/56354599-7b3bfa00-61d4-11e9-81fc-fdd84af700bf.png
[fi-with-ns-with-ip-addr]: https://user-images.githubusercontent.com/5430733/56358403-c8bd6480-61de-11e9-8f89-37281e9c7c86.png

**Comment by matsduf** [comment]
Fixed by #43
